src/Proxy.NET/Endpoints/Attendee.cs

```diff
@@ -0,0 +1,60 @@
+using System.Net;
+using System.Text.Json;
+using Proxy.NET.Models;
+using Proxy.NET.Services;
+
+namespace Proxy.NET.Endpoints;
+
+public static class Attendee
+{
+    private enum RequestType
+    {
+        AttendeeAdd,
+        AttendeeGetKey
+    }
+
+    public static void AttendeeEndpoints(this IEndpointRouteBuilder routes)
+    {
+        MapRoute.Post(routes, RequestType.AttendeeAdd, AttendeeAdd, Auth.Type.Jwt, "/attendee/event/{event_id}/register");
+        MapRoute.Get(routes, RequestType.AttendeeGetKey, AttendeeGetKey, Auth.Type.Jwt, "/attendee/event/{event_id}");
+    }
+
+    /// <summary>
+    /// Represents an asynchronous operation that can return a value.
+    /// </summary>
+    /// <returns>A task representing the asynchronous operation.</returns>
+    private static async Task AttendeeAdd(HttpContext context, RequestType requestType, string extPath)
+    {
+        var attendeeService = context.RequestServices.GetRequiredService<IAttendeeService>();
+        if (context.Items["RequestContext"] is not string userId || string.IsNullOrEmpty(userId))
+            throw new ArgumentException("Request context not found");
+        if (context.GetRouteData().Values["event_id"] is not string eventId || string.IsNullOrEmpty(eventId))
+            throw new ArgumentException("Event ID not found");
+
+        string api_key = await attendeeService.AddAttendeeAsync(userId, eventId);
+        context.Response.StatusCode = (int)HttpStatusCode.Created;
+        context.Response.ContentType = "application/json";
+        await context.Response.WriteAsync(JsonSerializer.Serialize(new { api_key }));
```

Comment:
We don't want to be handling serialization ourselves, we should delegate that up to the ASP.NET pipeline, instead we should either return an object or an explicit result type using [`TypedResults`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.typedresults?view=aspnetcore-8.0).

So the return type of the method shouldn't be `Task` but `Task<IResult>` and the code would look like:


```suggestion
        string api_key = await attendeeService.AddAttendeeAsync(userId, eventId);
        return TypedResults.OkObjectResult(new { api_key });
```

Reply:
fixed, nice!!